services/connector/commands/request_accounts.go

```diff
@@ -0,0 +1,88 @@
+package commands
+
+import (
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/status-im/status-go/rpc"
+	persistance "github.com/status-im/status-go/services/connector/database"
+)
+
+// errors
+var (
+	ErrRequestMissingDAppData      = errors.New("request missing dApp data")
+	ErrAccountsRequestDeniedByUser = errors.New("accounts request denied by user")
+)
+
+type RequestAccountsCommand struct {
+	rpcClient *rpc.Client
+	db        *sql.DB
+}
+
+type RequestAccountsResponse struct {
+	Accounts []string `json:"accounts"`
+}
+
+func (c *RequestAccountsCommand) requestAccountForDApp() (string, error) {
+	// NOTE: this is temporary implementation, actual code should invoke popup on the UI
+
+	// TODO: emit a request accounts signal and hang on wallet response
+	if false {
+		return "", ErrAccountsRequestDeniedByUser
+	}
+
+	accountsRequest := RPCRequest{
+		Method: "eth_accounts",
+		Params: []interface{}{},
+	}
+
+	requestJSON, err := json.Marshal(accountsRequest)
+	if err != nil {
+		return "", fmt.Errorf("failed to marshal request: %v", err)
+	}
+
+	return c.rpcClient.CallRaw(string(requestJSON)), nil
+}
+
+func (c *RequestAccountsCommand) Execute(request RPCRequest) (string, error) {
+	if request.DAppUrl == "" || request.DAppName == "" {
+		return "", ErrRequestMissingDAppData
+	}
+
+	dApp, err := persistance.SelectDAppByUrl(c.db, request.DAppUrl)
+	if err != nil {
+		return "", err
+	}
+
+	dApp.Name = request.DAppName
+	dApp.IconURL = request.DAppIconUrl
+	if dApp.URL == "" {
+		dApp.URL = request.DAppUrl
+	}
+
+	if dApp.SharedAccount == "" {
+		account, err := c.requestAccountForDApp()
+		if err != nil {
+			return "", err
+		}
+		dApp.SharedAccount = account
+		println("----------> Shared account for dApp: ", dApp.SharedAccount)
```

Comment:
some leftover here ?

Reply:
it's draft so i left some logs for testing, will remove them a bit later